fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/SkewJoinOptimizeRule.java

```diff
@@ -85,14 +89,91 @@ public class SkewJoinOptimizeRule extends TransformationRule {
     private static final Logger LOG = LogManager.getLogger(SkewJoinOptimizeRule.class);
 
     private static final String RAND_COL = "rand_col";
+
     public SkewJoinOptimizeRule() {
         super(RuleType.TF_SKEW_JOIN_OPTIMIZE_RULE,
                 Pattern.create(OperatorType.LOGICAL_JOIN, OperatorType.PATTERN_LEAF, OperatorType.PATTERN_LEAF));
     }
 
     @Override
     public boolean check(OptExpression input, OptimizerContext context) {
-        return ((LogicalJoinOperator) input.getOp()).getJoinHint().equals(JoinOperator.HINT_SKEW);
+        // respect the join hint
+        if (((LogicalJoinOperator) input.getOp()).getJoinHint().equals(JoinOperator.HINT_SKEW)) {
+            return true;
+        }
+
+        if (!context.getSessionVariable().isUseStatsToOptimizeSkewJoin()) {
```

Comment:
it is better to support both manual and statistics based modes.

Reply:
Hint is not manual?